theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/BasicKeyframedTrack.tsx

```diff
@@ -1,37 +1,44 @@
+import React, {useState} from 'react'
 import type {TrackData} from '@theatre/core/projects/store/types/SheetState_Historic'
-import type {SequenceEditorPanelLayout} from '@theatre/studio/panels/SequenceEditorPanel/layout/layout'
+import type {
+  DopeSheetSelection,
+  SequenceEditorPanelLayout,
+} from '@theatre/studio/panels/SequenceEditorPanel/layout/layout'
 import type {SequenceEditorTree_PrimitiveProp} from '@theatre/studio/panels/SequenceEditorPanel/layout/tree'
 import {usePrism} from '@theatre/react'
 import type {Pointer} from '@theatre/dataverse'
 import {val} from '@theatre/dataverse'
-import React from 'react'
-import styled from 'styled-components'
 import KeyframeEditor from './KeyframeEditor/KeyframeEditor'
-
-const Container = styled.div``
+import type {StrictRecord} from '@theatre/shared/utils/types'
+import isEqual from 'lodash-es/isEqual'
 
 const BasicKeyframedTrack: React.FC<{
   leaf: SequenceEditorTree_PrimitiveProp
-
   layoutP: Pointer<SequenceEditorPanelLayout>
   trackData: TrackData
 }> = React.memo(({layoutP, trackData, leaf}) => {
-  const {selectedKeyframeIds, selection} = usePrism(() => {
+  const [selection, setSelection] = useState<undefined | DopeSheetSelection>(
+    undefined,
+  )
+  const [selectedKeyframeIds, setSelectedKeyframeIds] = useState<
+    undefined | StrictRecord<string, true>
+  >({})
+
+  usePrism(() => {
```

Comment:
Was this component being invalidated too frequently?

Reply:
Yeah, and it still is actually. Let's discuss on a call in a bit

Reply:
Issue created https://github.com/theatre-js/theatre/issues/77